pkg/tlsconfig/tls_config.go

```diff
@@ -0,0 +1,109 @@
+package tlsconfig
+
+import (
+	"crypto/tls"
+	"fmt"
+	"strings"
+)
+
+type Config struct {
+	Ciphers    []string `toml:"ciphers"`
+	MinVersion string   `toml:"min-version"`
+	MaxVersion string   `toml:"max-version"`
+}
+
+func NewConfig() Config {
+	return Config{}
+}
+
+func (c Config) Validate() error {
+	_, err := c.Parse()
+	return err
+}
+
+func (c Config) Parse() (out *tls.Config, err error) {
+	if len(c.Ciphers) > 0 {
+		if out == nil {
```

Comment:
Would `out` ever be non-nil here? But I guess that's fine for consistency with the other blocks below.

Reply:
True. This is more resilient around future refactorings that may add a condition above it. Yea or nay?